File: src/main/java/wraith/fabricaeexnihilo/compatibility/rei/barrel/CompostDisplay.java
package wraith.fabricaeexnihilo.compatibility.rei.barrel;

import me.shedaniel.rei.api.common.category.CategoryIdentifier;
import me.shedaniel.rei.api.common.display.Display;
import me.shedaniel.rei.api.common.entry.EntryIngredient;
import me.shedaniel.rei.api.common.util.EntryIngredients;
import net.minecraft.util.Identifier;
import wraith.fabricaeexnihilo.compatibility.rei.PluginEntry;
import wraith.fabricaeexnihilo.recipe.barrel.CompostRecipe;

import java.util.List;
import java.util.Optional;

public class CompostDisplay implements Display {
    public final EntryIngredient input;
    public final EntryIngredient result;
    private final Identifier id;

    public CompostDisplay(CompostRecipe recipe) {
        this.input = EntryIngredients.ofIngredient(recipe.getInput());
        this.result = EntryIngredients.of(recipe.getResult());
        this.id = recipe.getId();
    }

    @Override
    public CategoryIdentifier<?> getCategoryIdentifier() {
        return PluginEntry.COMPOSTING;
    }

    @Override
    public List<EntryIngredient> getInputEntries() {
        return List.of(input);
    }

    @Override
    public List<EntryIngredient> getOutputEntries() {
        return List.of(result);
    }

    @Override
    public Optional<Identifier> getDisplayLocation() {
        return Optional.of(id);
    }
}

File: src/main/kotlin/exnihilofabrico/compatibility/rei/barrel/CompostDisplay.kt
package exnihilofabrico.compatibility.rei.barrel

import exnihilofabrico.compatibility.rei.PluginEntry
import exnihilofabrico.modules.ModBlocks
import exnihilofabrico.util.asREIEntry
import me.shedaniel.rei.api.RecipeDisplay

class CompostDisplay(val recipe: REICompostRecipe): RecipeDisplay {
    override fun getRecipeCategory() = PluginEntry.COMPOSTING

    override fun getOutputEntries() = listOf(recipe.reiOutput())
    override fun getInputEntries() =
        listOf(recipe.reiInputs(), ModBlocks.BARRELS.values.map { it.asREIEntry() })

}